### PR TITLE
initial wrapper

### PR DESCRIPTION
**Pull request by 7echno**
This is initial wrapper for the "verify" functions, with the few changes to verify.py file itself.

### CODE REVIEW

File: eccp [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        section = str(int(section) + 1)
+        run=True
+    else:
+        print "-> end of sections"
+        run=False
+        
+try:
+    if n % 2 == 0:
+        raise ValueError("not prime")
+    for i in xrange(3, int(n**0.5)+1, 2):
```
Comment (tomato42) [inline eccp]: xrange is not present in Python 3, please add an alias that will make it work on py3

File: eccp [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        print params
+# TODO: add some clean function to the end of the section, clean values in params
+        if set(c_t1) == set(par_list):
+            print "start ----> c_t1"
+            s = int(str(params['s']),16)
+            w = int(str(params['w']),16)
+            t = int(str(params['t']),16)
+            j = int(str(params['j']),16)
+            a = (3 * j * (oneseventwoeight - j)) % n
+            b = (2 * j * (oneseventwoeight - j)**2) % n
```
Comment (tomato42) [inline eccp]: it would be cleaner to create a method in the main file that will calculate those values by itself

File: eccp [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+#oneseventwoeight = 1728
+
+with open(sys.argv[1], 'r+') as f:
+    config_buff = f.read()
+    f.close()
+config_ini = StringIO.StringIO(config_buff)
+
+config = ConfigParser.RawConfigParser(allow_no_value=True)
+#config.read(config_ini)
```
Comment (tomato42) [inline eccp]: unused code, please remove

File: eccp [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+par_set()
+
+#s = int(str(params['s']),16)
+#w = int(str(params['w']),16)
+#j = int(str(params['j']),16)
+#t = int(str(params['t']),16)
+#a = int(str(params['a']),16)
+#b = int(str(params['b']),16)
+#q = int(str(params['q']),16)
+#r = int(str(params['r']),16)
```
Comment (tomato42) [inline eccp]: unused code, please remove

File: eccp [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+config = ConfigParser.RawConfigParser(allow_no_value=True)
+#config.read(config_ini)
+config.readfp(config_ini)
+sections = config.sections()
+
+# dict with params
+
+def par_set():
+    global params
```
Comment (tomato42) [inline eccp]: why `params` is a global?

File: verify.py [hunk 6]
```diff
@@ -153,6 +148,7 @@ def brillhart_lehmer_selfridge_test(n, s, b):
         raise ValueError("Invalid certificate")
     if pow(b, n-1, n) != 1:
         raise ValueError("Invalid certificate")
+#    if gcd(b ** s - 1, n) != 1:
```
Comment (tomato42) [inline verify.py]: unused code

File: eccp [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    else:
+        print "-> end of sections"
+        run=False
+        
+try:
+    if n % 2 == 0:
+        raise ValueError("not prime")
+    for i in xrange(3, int(n**0.5)+1, 2):
+        if n % i == 0:
+            raise ValueError("not prime: " + str(n))
```
Comment (tomato42) [inline eccp]: this whole trial division should be put in a function

File: eccp [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+print n
+print "==============="
+
+section = str(1)
+run=True
+while run: 
+    if section in sections:
+        par_list = []
+        options = config.options(section)
+        print "==============="
```
Comment (tomato42) [inline eccp]: python 3 incompatible syntax

File: eccp (hunk 5 above)
Comment (7echno) [inline eccp]: I am using it in latter in the code, and also resetting with new call to this function, maybe I can find a more elegant solution for this.

File: eccp (hunk 3 above)
Comment (7echno) [inline eccp]: removed

File: eccp (hunk 4 above)
Comment (7echno) [inline eccp]: removed

File: eccp (hunk 8 above)
Comment (7echno) [inline eccp]: fixed

File: eccp (hunk 2 above)
Comment (7echno) [inline eccp]: fixed

File: eccp (hunk 1 above)
Comment (7echno) [inline eccp]: fixed

File: eccp (hunk 7 above)
Comment (7echno) [inline eccp]: fixed

File: verify.py (hunk 6 above)
Comment (7echno) [inline verify.py]: removed

File: ecpp [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+# TODO: add some clean function to the end of the section, clean values in params
+        if set(c_t1) == set(par_list):
+            print("start ----> c_t1")
+            s = int(str(params['s']),16)
+            w = int(str(params['w']),16)
+            t = int(str(params['t']),16)
+            j = int(str(params['j']),16)
+            a = a_calc(j,n)
+            b = b_calc(j,n)
+            n = curve_test(n, s, w, a, b, t)
```
Comment (tomato42) [inline ecpp]: I thought of adding a new `curve_test`-like function that would hide the need to call `a_calc` and `b_calc` from calling code

File: ecpp [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+    else:
+        print("-> end of sections")
+        run = False
+
+       
+try:
+    if n % 2 == 0:
+        raise ValueError("not prime")
+    for i in xrange(3, int(n**0.5)+1, 2):
+        division(n,i)
```
Comment (tomato42) [inline ecpp]: I said "this whole trial division", meaning from line 123 till 126 😄

File: ecpp [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+config.readfp(config_ini)
+sections = config.sections()
+
+par_set()
+
+# combinations
+c_t1 = ['s','w','j','t']
+c_t2 = ['s','w','a','b','t']
+llr_t = ['s','q']
+bls_t = ['s','b']
```
Comment (tomato42) [inline ecpp]: whenever any of the above 4 constants is used, a new `set` is created from them, it's wasteful, they should be initialised as sets from beginning

File: ecpp [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+
+# prime candidate
+n = config.get('Candidate','N')
+print(n)
```
Comment (tomato42) [inline ecpp]: the code is printing `n` twice, once here and then in line 50, the difference is that here it doesn't say why or what it is printing...

File: ecpp [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+        par_list = []
+        options = config.options(section)
+        print("===============")
+        print(section)
+        print("===============")
+        for option in options:
+            print(option + '=' + config.get(section,option))
+            value = config.get(section,option)
+            # if starts with minus
+            params[str(option)] = value.replace("$","")
```
Comment (tomato42) [inline ecpp]: why the `str` here?

File: ecpp [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+            par_list.append(option)
+        print(par_list)
+        print(params)
+# TODO: add some clean function to the end of the section, clean values in params
+        if set(c_t1) == set(par_list):
+            print("start ----> c_t1")
+            s = int(str(params['s']),16)
```
Comment (tomato42) [inline ecpp]: why `str()` here, and in following lines?

also nit: there should be space after comma

File: ecpp (hunk 9 above)
Comment (7echno) [inline ecpp]: What do you think on extending curve_test functions with j argument, put calculation of a and b inside function, and then just change the call of it like : curve_test(n, s, w, None, None, t, j) or lattercurve_test(n, s, w, a, b, t, None) ?

File: ecpp (hunk 10 above)
Comment (7echno) [inline ecpp]: fixed :)

File: ecpp (hunk 12 above)
Comment (7echno) [inline ecpp]: fixed.

File: ecpp (hunk 9 above)
Comment (tomato42) [inline ecpp]: it would be better to change a, b and j to parameters with default and move those `None`s to the function definition

File: ecpp [hunk 15]
```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/python
+
+from __future__ import print_function
+import ConfigParser
+import sys
+import StringIO
+from verify import *
+
+#oneseventwoeight = 1728
```
Comment (tomato42) [inline ecpp]: unused

File: ecpp [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+
+# dict with params
+params = {}
+def par_set():
+    params.update({'s':None, 'w':None, 'j':None, 't':None, 'a':None, 'b':None, 'q':None, 'r':None})
+
+# division function
+def division(n):
```
Comment (tomato42) [inline ecpp]: the method is a library method, so it should be in `verify.py`, and second, it checks primality of the argument, provided the argument is small enough, so it probably should be something like `is_prime_64`

File: ecpp [hunk 17]
```diff
@@ -0,0 +1,122 @@
...
+            a = params['a']
+            b = params['b']
+            q = params['q']
+            if c_t1 == par_list:
+                print("start ----> c_t1")
+                n = curve_test(n, s, w, t, j=j)
+                print("n:")
+                print(n)
+                print("end ------> c_t1")
+                par_set()
```
Comment (tomato42) [inline ecpp]: why not call `par_set()` right after `while run:`? then only one call is needed...

**Comment by tomato42** [comment]
good enough for now, thanks!

**Pull request merged**